Warning : Error: Can't resolve 'domain' in '/node_modules/@sentry/hub/esm'

<!-- Requirements: please go through this checklist before opening a new issue -->

- [x] Review the documentation: https://docs.sentry.io/
- [x] Search for existing issues: https://github.com/getsentry/sentry-javascript/issues
- [x] Use the latest release: https://github.com/getsentry/sentry-javascript/releases
- [x] Provide a link to the affected event from your Sentry account

## Package + Version

- [x] `@sentry/browser`
- [ ] `@sentry/node`
- [ ] `raven-js`
- [ ] `raven-node` _(raven for node)_
- [ ] other:

### Version:

```
~5.15.2 + 5.15.3 (I don't have that warn with the 5.15.0)
And, I use angular 9.0 with ivy
```

## Description

When I build the app I get a nice ⚠️  WARNING
if I set the version of @sentry/browser to 5.15.0 I don't have it

```
chunk {4} styles.91325c2227fce484ed88.css (styles) 643 kB [initial] [rendered]
Date: 2020-03-27T10:52:35.191Z - Hash: 54ece857c5cbd64ddfe7 - Time: 208747ms

WARNING in /Users/tavenier/workspace/bridge-bo-portal/node_modules/@sentry/hub/esm/hub.js
Module not found: Error: Can't resolve 'domain' in '/Users/tavenier/workspace/bridge-bo-portal/node_modules/@sentry/hub/esm'
resolve 'domain' in '/Users/tavenier/workspace/bridge-bo-portal/node_modules/@sentry/hub/esm'
  Parsed request is a module
  using description file: /Users/tavenier/workspace/bridge-bo-portal/node_modules/@sentry/hub/package.json (relative path: ./esm)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      looking for modules in /Users/tavenier/workspace/bridge-bo-portal
        using description file: /Users/tavenier/workspace/bridge-bo-portal/package.json (relative path: .)
          Field 'browser' doesn't contain a valid alias configuration
          using description file: /Users/tavenier/workspace/bridge-bo-portal/package.json (relative path: ./domain)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /Users/tavenier/workspace/bridge-bo-portal/domain doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /Users/tavenier/workspace/bridge-bo-portal/domain.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /Users/tavenier/workspace/bridge-bo-portal/domain.tsx doesn't exist
            .mjs
              Field 'browser' doesn't contain a valid alias configuration
              /Users/tavenier/workspace/bridge-bo-portal/domain.mjs doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /Users/tavenier/workspace/bridge-bo-portal/domain.js doesn't exist
            as directory
              /Users/tavenier/workspace/bridge-bo-portal/domain doesn't exist
 
...
```

Same here, same version
"@sentry/browser": "^5.15.3",

WARNING in ./node_modules/@sentry/hub/esm/hub.js
Module not found: Error: Can't resolve 'domain' in '/media/marco/480gb/angular/olser/node_modules/@sentry/hub/esm'

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,10 @@
 
 - "You miss 100 percent of the chances you don't take. — Wayne Gretzky" — Michael Scott
 
+## 5.15.4
+
+- [node] fix: Path domain onto global extension method to not use require (#2527)
+
 ## 5.15.3
 
 - [hub] fix: Restore dynamicRequire, but for `perf_hooks` only (#2524)

--- a/packages/hub/src/hub.ts
+++ b/packages/hub/src/hub.ts
@@ -454,8 +454,14 @@ export function getCurrentHub(): Hub {
  */
 function getHubFromActiveDomain(registry: Carrier): Hub {
   try {
-    const req = require;
-    const domain = req('domain');
+    const property = 'domain';
+    const carrier = getMainCarrier();
+    const sentry = carrier.__SENTRY__;
+    // tslint:disable-next-line: strict-type-predicates
+    if (!sentry || !sentry.extensions || !sentry.extensions[property]) {
+      return getHubFromCarrier(registry);
+    }
+    const domain = sentry.extensions[property] as any;
     const activeDomain = domain.active;
 
     // If there no active domain, just return global hub

--- a/packages/hub/src/hub.ts
+++ b/packages/hub/src/hub.ts
@@ -17,16 +17,6 @@ import { consoleSandbox, getGlobalObject, isNodeEnv, logger, timestampWithMs, uu
 import { Carrier, Layer } from './interfaces';
 import { Scope } from './scope';
 
-declare module 'domain' {
-  export let active: Domain;
-  /**
-   * Extension for domain interface
-   */
-  export interface Domain {
-    __SENTRY__?: Carrier;
-  }
-}
-
 /**
  * API compatibility version of this hub.
  *

--- a/packages/node/src/index.ts
+++ b/packages/node/src/index.ts
@@ -39,6 +39,8 @@ export { defaultIntegrations, init, lastEventId, flush, close } from './sdk';
 export { SDK_NAME, SDK_VERSION } from './version';
 
 import { Integrations as CoreIntegrations } from '@sentry/core';
+import { getMainCarrier } from '@sentry/hub';
+import * as domain from 'domain';
 
 import * as Handlers from './handlers';
 import * as NodeIntegrations from './integrations';
@@ -50,3 +52,14 @@ const INTEGRATIONS = {
 };
 
 export { INTEGRATIONS as Integrations, Transports, Handlers };
+
+// We need to patch domain on the global __SENTRY__ object to make it work for node
+// if we don't do this, browser bundlers will have troubles resolving require('domain')
+const carrier = getMainCarrier();
+if (carrier.__SENTRY__) {
+  carrier.__SENTRY__.extensions = carrier.__SENTRY__.extensions || {};
+  if (!carrier.__SENTRY__.extensions.domain) {
+    // @ts-ignore
+    carrier.__SENTRY__.extensions.domain = domain;
+  }
+}

--- a/packages/node/src/sdk.ts
+++ b/packages/node/src/sdk.ts
@@ -101,7 +101,7 @@ export function init(options: NodeOptions = {}): void {
     options.environment = process.env.SENTRY_ENVIRONMENT;
   }
 
-  if (domain.active) {
+  if ((domain as any).active) {
     setHubOnCarrier(getMainCarrier(), getCurrentHub());
   }
 

--- a/packages/node/test/domain.test.ts
+++ b/packages/node/test/domain.test.ts
@@ -1,6 +1,10 @@
 import { getCurrentHub, Hub } from '@sentry/core';
 import * as domain from 'domain';
 
+import * as Sentry from '../src';
+
+console.log(Sentry.SDK_NAME);
+
 describe('domains', () => {
   test('without domain', () => {
     expect(domain.active).toBeFalsy();

--- a/packages/node/test/domain.test.ts
+++ b/packages/node/test/domain.test.ts
@@ -1,8 +1,10 @@
 import { getCurrentHub, Hub } from '@sentry/core';
 import * as domain from 'domain';
 
+// We need this import here to patch domain on the global object
 import * as Sentry from '../src';
 
+// tslint:disable-next-line: no-console
 console.log(Sentry.SDK_NAME);
 
 describe('domains', () => {

